src/validation/schema.js

```diff
@@ -339,10 +582,39 @@ const validateTypeDefinition = (defs, def) =>
 const validateTypeDefinitions = defs =>
   defs.reduce((errors, def) => errors.concat(validateTypeDefinition(defs, def)), List())
 
+const validateNamingCollisionsInTypes = types => {
+  let seen = Set()
+  let errored = Set()
+  return types.reduce((errors, type) => {
+    if (seen.has(type) && !errored.has(type)) {
+      errors = errors.push(
+        immutable.fromJS({
+          loc: { start: 1, end: 1 },
+          entity: type,
+          message: `Type ${type} is defined more than once`,
+        }),
+      )
+      errored = errored.add(type)
+    } else {
+      seen = seen.add(type)
+    }
+    return errors
+  }, List())
+}
+
+const validateNamingCollisions = (local, foreign) =>
+  List.of(...validateNamingCollisionsInTypes(local.concat(foreign)))
```

Comment:
This could just return `validateNamingCollisionsInTypes(...)` without the `List.of(...)`